### PR TITLE
replace str.decode / str.encode for base64 and quoted-printable. should fix #20

### LINKED ISSUE #20: Make vobject python3-compatible
althrough vobject is distributed as python3-compatible, it contains lines like https://github.com/eventable/vobject/blob/master/vobject/vcard.py#L137 - str.decode("base64") etc is only supported in python2, so using vobject in a python3-environment especially fails when base64-encoded photos or such are parsed.

### PR DESCRIPTION
**Pull request by prauscher**
as said. assumed internal encoding of utf-8, but actually i'm not entirely sure if this is how you handle things. therefore please tell me what to change and i'll do extra commits

### CODE REVIEW

**Comment by arska** [comment]
@prauscher I had to add bytes() before base64 decoding, see https://github.com/arska/vobject/commit/fd07a9f2f06f5862699982ee47d5d789ac7924a2

**Comment by prauscher** [comment]
@arska thank you, but line.value.encode("utf-8") works in python2 too

**Comment by serathius** [comment]
+1

File: vobject/vcard.py [hunk 1]
```diff
@@ -147,7 +149,7 @@ def encode(cls, line):
         if not line.encoded:
             encoding = getattr(line, 'encoding_param', None)
             if encoding and encoding.upper() == cls.base64string:
-                line.value = line.value.encode('base64').replace('\n', '')
+                line.value = codecs.encode(line.value.encode(coding), "base64").decode("utf-8")
```
Comment (PrashantSinghal) [inline vobject/vcard.py]: The variable 'coding' is not defined? Is this a typo?
It looks like when my vcard has a photo which is base64 encoded, this breaks vcard.serialize().

I'm attaching a sample vcard test_file. The vcard has photo data encoded in base64 format.
A simple vcard.readOne(...).serialize() will fail.
[sample_vcard.txt](https://github.com/eventable/vobject/files/591494/sample_vcard.txt)

Comment (pbiering) [inline vobject/vcard.py]: Perhaps my pull request on top will fix this: https://github.com/eventable/vobject/pull/49/files

**Pull request merged**